docs/logging.md

```diff
@@ -0,0 +1,134 @@
+# Logging
+
+Thanos supports request logging via flags.
+
+Components are configured using `--request.logging-config-file` to reference to the configuration file or `--request.logging-config` as YAML directly.
+
+## Configuration
+
+Configuration can be supplied for either gRPC, HTTP or both. Options can be supplied globally, which is applied to both gRPC and HTTP, or individually to either or both protocols.
+
+### Options
+
+Valid `level` for `options` should be one of `INFO`, `WARNING`, `ERROR` or `DEBUG`.
```

Comment:
So what does `level` set? :thinking:

Reply:
So `level`, from what I can see provides an additional filter. Take the following example

I have enabled `--log.level=debug` globally and request logging is enabled with `ERROR` - I will not see any of the `debug` logs emitted from the http middleware.

This could be seen as a nuisance or as a feature :)

Reply:
Understood, then let's keep it!